test/integration/access_log_integration_test.cc

```diff
@@ -0,0 +1,27 @@
+#include "test/integration/http_integration.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+using testing::HasSubstr;
+
+namespace Envoy {
+
+class AccessLogIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                                 public HttpIntegrationTest {
+public:
+  AccessLogIntegrationTest() : HttpIntegrationTest(Http::CodecType::HTTP1, GetParam()) {}
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, AccessLogIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+TEST_P(AccessLogIntegrationTest, DownstreamDisconnectBeforeHeadersResponseCode) {
```

Comment:
I think it would also be good to add a positive test, to validate that status is set in the stream info.

Reply:
@yanavlasov can you please clarify what test it is you are suggesting?

Reply:
@yanavlasov ?